File: Source/AlfredBackend/Models/BotSettings.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AlfredBackend.Models
{
    public class BotSettings
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public string TwitchUserId { get; set; } = string.Empty; // FK to TwitchUser
        
        public List<ComponentSetting> Components { get; set; } = new();
        public List<TimeoutSetting> Timeouts { get; set; } = new();
        public string ConnectionStatus { get; set; } = "Disconnected";
        public string? ChannelName { get; set; }
        
        // Navigation
        [ForeignKey(nameof(TwitchUserId))]
        public TwitchUser User { get; set; } = null!;
    }
}

File: Source/frontend/src/app/models/bot-settings.ts
export interface ComponentSetting {
    name: string;
    description: string;
    enabled: boolean;
}

export interface TimeoutSetting {
    name: string;
    description: string;
    valueSeconds: number;
}

export interface BotSettings {
    components: ComponentSetting[];
    timeouts: TimeoutSetting[];
    connectionStatus: string;
    channelName?: string;
}
